src/jadhav_lab_to_nwb/olson_2024/olson_2024_spike_gadgets_lfp_interface.py

```diff
@@ -0,0 +1,178 @@
+                        "properties": {
+                            "name": {"type": "string"},
+                            "description": {"type": "string"},
+                        },
+                    },
+                },
+            },
+        }
+
+        return metadata_schema
+
+    def add_to_nwbfile(self, nwbfile: NWBFile, metadata: dict, stub_test: bool = False):
+        folder_path = Path(self.source_data["folder_path"])
+        electrodes_table = nwbfile.electrodes.to_dataframe()
+        lfp_data, lfp_electrodes, conversions = [], [], []
+        for file_path in folder_path.glob("*.dat"):
+            if file_path.name.startswith("._") or file_path.stem == "SL18_D19.timestamps":
+                continue
+            fieldsText = readTrodesExtractedDataFile(file_path)
+            data = fieldsText["data"]
+            conversion = float(fieldsText["voltage_scaling"]) * 1e-6
+            channel_number = file_path.stem.split("ch")[-1]
+            trode_number = file_path.stem.split("ch")[0].split("nt")[-1]
+            chID = f"nTrode{trode_number}_elec{channel_number}"
+            channel_index = electrodes_table.index[electrodes_table.chID == chID][0]
+            assert electrodes_table.hasLFP[
+                channel_index
+            ], f"Channel {chID} has LFP data, but is not marked as an LFP channel."
+            lfp_electrodes.append(channel_index)
+            lfp_data.append(data)
+            conversions.append(conversion)
+        timestamp_file_path = folder_path / "SL18_D19.timestamps.dat"
+        fieldsText = readTrodesExtractedDataFile(timestamp_file_path)
+        timestamps = np.asarray(fieldsText["data"], dtype=np.float64)
+        for conversion in conversions:
+            assert conversion == conversions[0], "All LFP data must have the same conversion factor."
+        conversion = conversions[0]
+        lfp_data = np.array(lfp_data, dtype=np.int16).T
+        lfp_table_region = nwbfile.create_electrode_table_region(
+            region=lfp_electrodes,
+            description="LFP electrodes",
+        )
+        if stub_test:
+            lfp_data = lfp_data[:100]
+            timestamps = timestamps[:100]
+        lfp_metadata = metadata["Ecephys"]["LFP"]
+        lfp_electrical_series = ElectricalSeries(
+            name=lfp_metadata["ElectricalSeries"][0]["name"],
+            description=lfp_metadata["ElectricalSeries"][0]["description"],
+            data=lfp_data,
+            timestamps=timestamps,
+            electrodes=lfp_table_region,
+            conversion=conversion,
+        )
+        lfp = LFP(electrical_series=lfp_electrical_series)
+        ecephys_module = nwb_helpers.get_module(
+            nwbfile,
+            name="ecephys",
+            description="Processed extracellular electrophysiology data.",
+        )
+        ecephys_module.add(lfp)
+
+
+def readTrodesExtractedDataFile(filename: FilePath) -> dict:
```

Comment:
Isn't it the same as the one in olson_2024_behavior_interface.py? 
Would you consider moving it to utils.py so both interfaces can use it?

Reply:
yeah, that's the plan (should've mentioned it). I just wanted to keep them separate first to avoid merge conflicts.